### PR TITLE
Lw/bugfixes

### PR DESCRIPTION
**Pull request by phi-lira**
### Purpose of this PR
Bugfixes:
 * Fixed specular issues on mobile. [case 1109017](https://issuetracker.unity3d.com/issues/scaled-objects-render-shadows-and-specularity-incorrectly-in-the-lwrp-on-device)
 * Fixed issue causing LWRP to create MSAA framebuffer even when MSAA setting is disabled.
 * Disabled post-processing in mobile as it was breaking entire rendering.

Add Quality Tiers per-platforms:
LWRP doesn't use shader quality tiers from Graphics Tiers. These are not exposed in SRP. We need our own mechanism of shader quality and only allow a shader quality variant per-tier/per-asset. There's work planned to look at exposing shader quality tiers in LWRP. At the moment this PR adds 3 hardcoded tiers:

* SHADER_QUALITY_HIGH: Any non mobile platform. This can be forced in any platforms by defining SHADER_QUALITY_HINT_NICE 1.
* SHADER_QUALITY_MEDIUM: Mobile Platforms except for GLES2.
* SHADER_QUALITY_LOW: GLES2.

About QualitySettings.antiAliasing:
It seems there's no way to request MSAA On/Off when rendering to backbuffer besides settings the QualitySettings. This was causing effect that even when in the pipeline settings and camera settings MSAA was disabled the pipeline would render to MSAA framebuffers if QualitySettings.antiAliasing > 1. 

This PR saves and restores QualitySettings.antiAliasing per camera and when rendering it checks camera data and change QualitySettings.antiAliasing depending on camera msaa data.

This is an interim solution for a bigger problem. QualitySettings.antiAliasing is deeply hooked up in GfxDevice affecting mostly mobile and VR. 

---
### Release Notes
 Added.

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=lw%2Fbugfixes&unity_branch=2019.1%2Fstaging

**Manual Tests**: Tested shader ALU performance on iPhone 6S on a scene that's know to be ALU bound. No visible performance regression is introduced by this PR.

**Automated Tests**: Added performance tests on another github project.

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: Medium. Changes how normalization happens in different platforms. 

**Halo Effect**: Medium. Affects differently multiple platforms.

---

### CODE REVIEW

File: com.unity.render-pipelines.lightweight/ShaderLibrary/Core.hlsl [hunk 1]
```diff
@@ -15,6 +15,21 @@
 #endif
 #endif
 
+// Shader Quality Tiers in LWRP. 
+// SRP doesn't use Graphics Settings Quality Tiers.
+// We should expose shader quality tiers in the pipeline asset.
+// Meanwhile, it's forced to be:
+// High Quality: Non-mobile platoforms or shader explicit defined SHADER_HINT_NICE_QUALITY
```
Comment (aleks01010101) [inline com.unity.render-pipelines.lightweight/ShaderLibrary/Core.hlsl]: platoforms -> platforms

File: com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs [hunk 2]
```diff
@@ -218,6 +221,10 @@ static void InitializeCameraData(LightweightRenderPipelineAsset settings, Camera
             cameraData.postProcessLayer = camera.GetComponent<PostProcessLayer>();
             cameraData.postProcessEnabled = cameraData.postProcessLayer != null && cameraData.postProcessLayer.isActiveAndEnabled;
 
+            // Disables postprocessing in mobile VR. It's stable on mobile yet.
+            if (cameraData.isStereoEnabled && Application.isMobilePlatform)
```
Comment (aleks01010101) [inline com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs]: Release notes in the bug say "disabled msaa on mobile platform", this says "disabled msaa on mobile platform in VR"

File: com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs [hunk 3]
```diff
@@ -209,6 +208,10 @@ static void InitializeCameraData(LightweightRenderPipelineAsset settings, Camera
                 cameraData.msaaSamples = (camera.targetTexture != null) ? camera.targetTexture.antiAliasing : settings.msaaSampleCount;
             else
                 cameraData.msaaSamples = 1;
+            
+            // We have to set antiAliasing settings as this is the only way to enable/disable MSAA for backbuffer
+            // We save/restore this settings per-camera in RenderSingleCamera.
+            QualitySettings.antiAliasing = cameraData.msaaSamples;
```
Comment (aleks01010101) [inline com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs]: This is potentially very slow.

File: com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs (hunk 2 above)
Comment (phi-lira) [reply com.unity.render-pipelines.lightweight/Runtime/LightweightRenderPipeline.cs]: Fixed in release notes.

**Pull request merged**